terraform/plan.go

```diff
@@ -50,6 +58,47 @@ func (p *Plan) String() string {
 	return buf.String()
 }
 
+// Sums returns a PlanSums with sums of resources to be modified
+// based on given Diff
+func (p *Plan) Sums() *PlanSums {
+	modules := p.Diff.Modules
+
+	toChange := 0
+	toCreate := 0
+	toDestroy := 0
+	toDestroyAndCreate := 0
```

Comment:
You can avoid the local vars by initializing the `PlanSums` struct empty here and incrementing its fields in the loop below, since they'll each start at zero.